src/ContainerTerminal.jsx

```diff
@@ -91,44 +111,72 @@ class ContainerTerminal extends React.Component {
-                        superuser: this.props.system ? "require" : null,
-                        binary: false
-                    };
-
-                    const channel = cockpit.channel(opts);
-                    channel.wait()
-                            .then(() => {
-                                // Show the terminal. Once it was shown, do not show it again but reuse the previous one
-                                if (!this.state.opened) {
-                                    this.state.term.open(this.refs.terminal);
-                                    this.setState({ opened: true });
-
-                                    self.state.term.onData(function(data) {
-                                        if (self.state.channel)
-                                            self.state.channel.send(data);
-                                    });
-                                }
-
-                                channel.addEventListener("message", this.onChannelMessage);
-                                channel.addEventListener('close', this.onChannelClose);
-
-                                channel.send(String.fromCharCode(12)); // Send SIGWINCH to show prompt on attaching
-                                this.setState({ channel: channel, control_channel: out.sockets.control_socket, errorMessage: "" });
-                                this.resize(this.props.width);
-                            })
-                            .catch(e => {
-                                let message = cockpit.format(_("Could not open channel: $0"), e.problem);
-                                if (e.problem === "not-supported")
-                                    message = _("This version of the Web Console does not support a terminal.");
-                                this.setState({ errorMessage: message });
-                            });
-                })
-                .catch(e => this.setState({ errorMessage: cockpit.format(_("Could not attach to this container: $0"), e.problem) }));
+        // FIXME: Implement proper terminal - if not tty then need to do 'exec' and connect to it
+        // Tracked in: https://github.com/cockpit-project/cockpit-podman/issues/297
+        // Blocked on https://github.com/containers/libpod/issues/5741
+        // POC in varlink API: https://github.com/marusak/cockpit-podman/tree/proper_terminal
+
+        const channel = cockpit.channel({
+            payload: "stream",
+            unix: client.getAddress(this.props.system),
+            superuser: this.props.system ? "require" : null,
+            binary: true
+        });
+
+        channel.send("POST /v1.24/libpod/containers/" + encodeURIComponent(this.state.container) +
+                      "/attach?&stdin=true&stdout=true&stderr=true HTTP/1.0\r\n" +
+                      "Content-Length: 0\r\n\r\n");
+
+        const buffer = channel.buffer();
+
+        buffer.callback = function(data) {
+            let ret = 0;
```

Comment:
This whole function needs some comments about what it's doing on a high level, even for your future self.

Reply:
Done